collections/binary_search_tree.ts

```diff
@@ -277,7 +274,8 @@ export class BinarySearchTree<T> implements Iterable<T> {
    * Returns true if found and removed.
    */
   remove(value: T): boolean {
-    return !!this.removeNode(value);
+    const node: BinarySearchNode<T> | null = this.findNode(value);
+    return !!(node && this.removeNode(node));
```

Comment:
I know I suggested this before, but lets assume that if a node is found, it will be removed.

```ts
const node: BinarySearchNode<T> | null = this.findNode(value);
if (node) this.removeNode(node);
return !!node;
```

Reply:
Maybe `node != null` instead of `!!node`